browser/src/control/jsdialog/Widget.IconView.ts

```diff
@@ -174,12 +174,76 @@ JSDialog.iconView = function (
 	data: IconViewJSON,
 	builder: JSBuilder,
 ) {
+	/*
+	    0. how does iconview send core "which" entry is selected
+		=> _defaultCallbackHandler is responsible. there's a callback `builder.callback('iconview', 'select'` which is set
+		   for on click on the entries. when this is fired, a select message is sent to core and it's processed there IIUC.
+		   so for the dialog, it's just that it has to pass on it's onclick to it's child ?? or  maybe adding a wrapper
+		   around the iconview was a mistake
+
+		  0.1 how does iconview refresh it's contents on scroll
+		  => each entry on creation sets up an OnDemandRenderer where internally IntersectionObserver observes the widget
+			 and when it becomes visible, render is called on it. i don't know what constructs the JSDialog.Dropdown has but
+			 i don't think it has anything close to this yet. TODO
+
+		1. resize the iconview dropdown, show more icons
+		  1.1 increase the height to show x4 the height of the preview
+		  1.2 show it over the original iconview
+
+		2. pass on keyboard/mouse events from dropdown to iconview
+
+		  2.1 on scroll, render the visible ones
+		  => on scroll, there's a log entry on the console stdout saying "unhandled action" render_entry
+		     in the dropdown and this should somehow be passed on to the child which has it implemented
+		     in some callback
+				if (requestRender) {
+					// render on demand
+					var onIntersection = (entries: any) => {
+						entries.forEach((entry: any) => {
+							if (entry.isIntersecting) {
+								builder.callback(
+									controlType,
+									'render_entry',
+									{ id: controlId },
+									entryId +
+									...
+			 this code is responsible for rendering icons when the iconview is scrolled,
+			 i wonder where's this 'render_entry' thing handled in code.
+
+			 this  callback calls _defaultCallbackHandler which then creates a string with
+			 some data and calls app.socket.sendMessage(message); on that, which then does what 
+
+		  2.2 do this on both mouse and keyboard events
+
+		  2.3 on click, apply the thing and close the dropdown
+	*/
+	const commonContainer = L.DomUtil.create(
+		'div',
+		builder.options.cssClass + ' ui-iconview-window',
+		parentContainer,
+	);
+	commonContainer.id = data.id;
+
 	const container = L.DomUtil.create(
 		'div',
 		builder.options.cssClass + ' ui-iconview',
-		parentContainer,
+		commonContainer,
 	);
-	container.id = data.id;
+	container.id = data.id + '-iconview';
+
+	/* this is the list of iconviews for which we do not add an
+	 * expander for various reasons (eg. one being on the sidebar) */
+	const noExpanderIconviews = ['layoutpanel_icons-iconview'];
```

Comment:
Best to not hardcode any ids which may be changed later.
Instead please add new field in the JSON (IconViewJSON), like: `hasExpander: boolean`
Then it will be responsibility of widget creator to put that property when needed.
I think it will be used only in the notebookbar.

Then in Control.NotebookbarBuilder.js we can override iconview handler and just add desired property to the JSON and call the original handler.